lib/rucio/core/replica.py

```diff
@@ -4156,3 +4155,74 @@ def get_RSEcoverage_of_dataset(scope, name, *, session: "Session"):
             result[rse_id] = total
 
     return result
+
+@transactional_session
+def refresh_replicas(rse_id=None, replicas=None, *, session: "Session"):
```

Comment:
For the type annotations failure, there are missing type annotations here:

```suggestion
def refresh_replicas(
        rse_id: Optional[str] = None, 
        replicas: Optional[Iterable[dict[str, Any]]] = None, 
        *, 
        session: "Session"
) -> bool:
```